Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2678 @@
+
+As documented at [Apple Developer Documentation](https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/EntitlementKeyReference/Chapters/EnablingAppSandbox.html#//apple_ref/doc/uid/TP40011195-CH4-SW19 "Adding an App to an App Group"), the App Groups entitlement is required to share information between different apps through IPC or a shared file container, which means that data can be shared on the device directly between the apps.
+This entitlement is also required if an app extension requires to [share information with its containing app](https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/ExtensionScenarios.html "Sharing Data with Your Containing App").
+
+Depending on the data to-be-shared it might be more appropriate to share it using another method such as through a back end were this data could be potentially verified, avoiding tampering by e.g. the user himself.
+
+##### Embedded Provisioning Profile File
+
+If not having the original source code project you should then analyze the IPA and search inside for the *embedded provisioning profile* that is usually located in the root app bundle folder (`Payload/<appname>.app/`) under the name `embedded.mobileprovision`.
+
+This file is not a `.plist`, it is encoded using [Cryptographic Message Syntax](https://en.wikipedia.org/wiki/Cryptographic_Message_Syntax). On macOS you can [inspect an embedded provisioning profile's entitlements](https://developer.apple.com/library/archive/technotes/tn2415/_index.html#//apple_ref/doc/uid/DTS40016427-CH1-PROFILESENTITLEMENTS) using the following command:
+
+```bash
+$ security cms -D -i embedded.mobileprovision
+```
+
+and then search for the Entitlements key region (`<key>Entitlements</key>`).
+
+##### Entitlements Embedded in the Compiled App Binary
+
+If you only have the compiled app (IPA) or even only the running app, you normally won't be able to find `.entitlements` files. This could be also the case for the `embedded.mobileprovision` file. Still, you'll be able to extract the entitlements files from the app bundle yourself.
+
+First you need to find the path to the app's bundle. This can be easily done with objection, example using Telegram:
+
+- open the app and leave it running on foreground
+- run the following command
+    ```bash
+    $ objection --gadget Telegram run env | grep BundlePath
+    BundlePath  /var/containers/Bundle/Application/15E6A58F-1CA7-44A4-A9E0-6CA85B65FA35/Telegram X.app
+    ```
+
+Now you have two options:
+
+- connect per SSH, `cd` to the bundle and locate the (encrypted) app binary and use grep directly on it
+- or decrypt and extract the binary to your computer and use other tools apart from grep like binwalk or radare2.
```

Comment:
You mean to clutch it (decrypt the binary)?
In both scenarios you would need a jailbroken device. Or is there a way to do it on a non jailbroken device? If not, please mention in this paragraph that this is only working on a jb device.

Reply:
With decrypt and extract I mean exactly that. I've added the tools that can be used for that like clutch and frida-ios-dump, however for me clutch is not an option anymore, it depends on iOS version compatibility, I've left a comment on that.

I've re-written the whole section "Entitlements Embedded in the Compiled App Binary" to make everything clearer. Now it has two subsections on accessing the binary and extracting the plists. I highlight now that mainly you would have the IPA (and you can already try to extract the plists, works when encrypted) or you need a jailbroken device.